src/hotspot/cpu/x86/vm_version_x86.cpp

```diff
@@ -2588,6 +2592,10 @@ void VM_Version::initialize() {
   uint64_t   CPUFeatures_x64 = CPUFeatures_parse(GLIBCFeatures_x64);
   uint64_t       features_missing =   CPUFeatures_x64 & ~      _features;
   uint64_t glibc_features_missing = GLIBCFeatures_x64 & ~_glibc_features;
+
+  // Workaround JDK-8311164: CPU_HT is set randomly on hybrid CPUs like Alder Lake.
+  features_missing &= ~CPU_HT;
```

Comment:
Note: I just realized these missing features are different from the others, it's more "extra features" -- they are specified via the option, but not available on the current CPU.

Reply:
I am not sure what you mean here. This part of the code checks that the `-XX:CPUFeatures=0x...,0x...` specifies only a subset of the current options, never superset. Or do you have a suspection there exist some other options besides `CPU_HT` which are not fatal to use even if the CPU does not support them?

Reply:
> This part of the code checks that the -XX:CPUFeatures=0x...,0x... specifies only a subset of the current options, never superset.

Good, that was my understanding as well. I mean just the variables names are the same, but the meaning is different. But I don't suppose it should be fixed. The PR is good.

Thanks!